### PR TITLE
make persistent service not to support project k for now.

### PR DESCRIPTION
**Pull request by heejaechang**
also change to use cached IVsHierarchy only for error list filtering due to issue with caching it on shared projects

### CODE REVIEW

**Comment by heejaechang** [comment]
@srivatsn @shyamnamboodiripad @mavasani can you take a look? this is to make error list support project k properly

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/LinkedFileUtilities.cs [hunk 1]
```diff
@@ -70,9 +70,7 @@ private IVsHierarchy GetSharedItemContextHierarchy(IVisualStudioHostDocument doc
                 return null;
             }
 
-            var sharedHierarchy = document.SharedHierarchy;
```
Comment (Pilchie) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/LinkedFileUtilities.cs]: It looks like `document.SharedHierarchy` should be removed, if you can't trust it's value.  Note that with asynchronous solution load there are cases in non-xproj shared projects where only one head may be loaded at the time the file is opened.

**Comment by Pilchie** [comment]
@dpoeschl

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/LinkedFileUtilities.cs (hunk 1 above)
Comment (heejaechang) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/LinkedFileUtilities.cs]: @Pilchie yes, we need to figure out when we cache the info. unfortunately we can't remove this due to supporting filtering in error list for project k. we can't ask the question from background thread using IVsHierarchy.

I made not to use the cache for existing code but use it only in error list. for RTM, we should figure out right way to cache this.

Comment (heejaechang) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/LinkedFileUtilities.cs]: @Pilchie by the way, even editor case, it is possible for LinedFileUtilties to answer no for shared file. not sure what would happen in those case.

**Comment by heejaechang** [comment]
@srivatsn @shyamnamboodiripad @mavasani @dpoeschl @Pilchie I would like to check this in for RC. can you take a look please?

**Comment by Pilchie** [comment]
I'm not very happy with this because I expect people will continue to use `SharedProject`, but we can revisit that as part of #1859.

**Comment by heejaechang** [comment]
@Pilchie this doesn't change behavior of shared project except error list won't filter shared project correctly. but I am not sure current behavior on error list filtering on shared project is right one or not either.

**Comment by Pilchie** [comment]
Well, you added `SharedHierarchy` in db3fdb639abbf1e5656d7c9e9e0646c1dcf73cbd.  What happened for shared projects before that?

**Comment by heejaechang** [comment]
@Pilchie error list will return IVsHierarchy of the project rather than IVsHierarch of the shared project.that was the previous behavior.

**Comment by Pilchie** [comment]
That's what we want.

**Comment by heejaechang** [comment]
@Pilchie that's what we return, but error list will use shared project's IVsHierarchy for shared file. so, if that is behavior we want, error list need to filter based on current context of a file in the editor regardless what we return.

**Comment by heejaechang** [comment]
@srivatsn @shyamnamboodiripad @mavasani @dpoeschl ping? if we are going to take this, I need to check this in and port to preview branch.

**Comment by srivatsn** [comment]
:+1:

**Comment by dpoeschl** [comment]
:+1:

Do you intend for #1859 to cover the "// TODO: we need better way to see whether a project is project k project or not."? If so, can you call that out explicitly in that bug? If not, can you file a bug for that issue too?

**Comment by heejaechang** [comment]
done

**Comment by shyamnamboodiripad** [comment]
:+1:

**Pull request merged**